src/ansys/dynamicreporting/core/adr_service.py

```diff
@@ -837,9 +855,14 @@ def get_report(self, report_name: str) -> Report:
         """
         if self.serverobj is None:
             self.logger.error("Error: no connection to any service")
-            return None
+            raise ConnectionToServiceError
         my_report = Report(service=self, report_name=report_name)

```

Comment:
To be consistent, should this return -> None and raise based on my_report?   Try not to mix return error and exceptions, it makes it more complicated to write correct code around this function.